examples/small-study/src/studySetup.js

```diff
@@ -0,0 +1,154 @@
+/* eslint no-unused-vars: ["error", { "varsIgnorePattern": "getStudySetup" }]*/
+
+/**
+ *  Overview:
+ *
+ *  - constructs a well-formatted `studySetup` for use by `browser.study.setup`
+ *  - mostly declarative, except that some fields are set at runtime
+ *    asynchronously.
+ *
+ *  Advanced features:
+ *  - testing overrides from preferences
+ *  - expiration time
+ *  - some user defined endings.
+ *  - study defined 'shouldAllowEnroll' logic.
+ */
+
+/** Base for studySetup, as used by `browser.study.setup`.
+ *
+ * Will be augmented by 'getStudySetup'
+ */
+const baseStudySetup = {
+  // used for activeExperiments tagging (telemetryEnvironment.setActiveExperiment)
+  activeExperimentName: browser.runtime.id,
+
+  // uses shield sampling and telemetry semantics.  Future: will support "pioneer"
+  studyType: "shield",
+
+  // telemetry
+  telemetry: {
+    // default false. Actually send pings.
+    send: true,
+    // Marks pings with testing=true.  Set flag to `true` before final release
+    removeTestingFlag: false,
+  },
+
+  // endings with urls
+  endings: {
+    /** standard endings */
+    "user-disable": {
+      baseUrls: [
+        "https://qsurvey.mozilla.com/s3/Shield-Study-Example-Survey/?reason=user-disable",
+      ],
+    },
+    ineligible: {
+      baseUrls: [
+        "https://qsurvey.mozilla.com/s3/Shield-Study-Example-Survey/?reason=ineligible",
+      ],
+    },
+    expired: {
+      baseUrls: [
+        "https://qsurvey.mozilla.com/s3/Shield-Study-Example-Survey/?reason=expired",
+      ],
+    },
+
+    /** Study specific endings */
+    "user-used-the-feature": {
+      baseUrls: ["https://qsurvey.mozilla.com/s3/Shield-Study-Example-Survey/"],
+      category: "ended-positive",
+    },
+    "hated-the-feature": {
+      baseUrls: [
+        "https://qsurvey.mozilla.com/s3/Shield-Study-Example-Survey/?reason=hated-the-feature",
+      ],
+      category: "ended-negative",
+    },
+  },
+
+  // Study branches and sample weights, overweighing feature branches
+  weightedVariations: [
+    {
+      name: "feature-active",
+      weight: 1.5,
+    },
+    {
+      name: "feature-passive",
+      weight: 1.5,
+    },
+    {
+      name: "control",
+      weight: 1,
+    },
+  ],
+
+  // maximum time that the study should run, from the first run
+  expire: {
+    days: 14,
```

Comment:
@gregglind Does this value have to be integer or can be in decimals to handle expiring study in less than 24 hours